### PR TITLE
Add string to list comparison and cython module

### PR DESCRIPTION
**Pull request by GustavoPeredo**
Hey, so I needed to compare a bunch of strings to a single string and I noticed this module didn't have this functionality.

Inspired by https://github.com/gfairchild/pyxDamerauLevenshtein, I implemented this feature and a Cython version of the code (so it can run faster! Speeeeed!)

Time comparisons:
Array comparison
```
>>> print(timeit.timeit('import distancec; distancec.get_jaro_distance_array("test", ["tes","yesy3","test"]);', number=100000))
2.8039388999999915
>>> print(timeit.timeit('import distance; distance.get_jaro_distance_array("test", ["tes","yesy3","test"]);', number=100000))
7.372891200000026
```
Single string comparison
```
>>> print(timeit.timeit('import distancec; distancec.get_jaro_distance("test", "tes");', number=100000))
0.9566795999999727
>>> print(timeit.timeit('import distance; distance.get_jaro_distance("test", "tes");', number=100000))
2.402404699999977
```

### CODE REVIEW

File: pyjarowinkler/distance.py [hunk 1]
```diff
@@ -37,6 +37,17 @@ def get_jaro_distance(first, second, winkler=True, winkler_ajustment=True, scali
 
     return jaro
 
+def get_jaro_distance_array(string, array, winkler = True, winkler_ajustment = True, scaling = 0.1):
```
Comment (nap) [inline pyjarowinkler/distance.py]: I would is something like:

```pyton
def get_jaro_distance(first, second, winkler = True, winkler_ajustment = True, scaling = 0.1):
    if not instanceof(second, list):
        second = [second]
    return [calculate_jaro_distance(first, word, winkler, winkler_ajustment, scaling) for word in second]
```

**Review by nap — CHANGES_REQUESTED** [review]

**Comment by nap** [comment]
It is a rather interesting speed improvement.
This said, there's a few things that would need to be updated before I merge.

* I'd like it of instead of `distancec` it would by `cydistance`.
* Update the README to reflect this, maybe include the example above as well.
* Is the `distancec.c` necessary? It looks like something that was generated, maybe we want to re-generate it at install? I'm not familiar with cython. But if we can leave it out of the repo, it would be best.
* Can you update the tests as well as tox to run the test against both cython and pure python implementation?
* Please review my comments in the code as well.

Thank you!

**Comment by GustavoPeredo** [comment]
All your comments were taken into consideration and the files have been updated.
Travis CI Job passing from latest build:
https://travis-ci.com/github/GustavoPeredo/jaro-winkler-distance/jobs/521655317

**Comment by nap** [comment]
I'll have to review the implementation and get a better understanding of why it gives a different distance. I'll give you an update when I get to it.

I also noticed that the unit test only run against the python implementation, they should run and pass against the cython implementation as well. Right now, I don't think they will pass because the distance is significantly different.

Can you also squash together all the commits.
Thanks!

**Comment by GustavoPeredo** [comment]
> I'll have to review the implementation and get a better understanding of why it gives a different distance. I'll give you an update when I get to it.

This Stackoverflow answer explains the reason well: [Question](https://stackoverflow.com/questions/25226084/cython-vs-python-different-results-within-scipy-optimize-fsolve). Basically python works with C's double precision in floats while Cython works with C's float precision [(float and double explained)](https://hackr.io/blog/float-vs-double)... I can update the code to make the results match, it might be a good idea.

> I also noticed that the unit test only run against the python implementation, they should run and pass against the cython implementation as well. Right now, I don't think they will pass because the distance is significantly different.

The reason why I didn't implement the unit tests for cython is because I declared some functions as C-only functions to gain better performance. It's also possible to change that, but I'd suggest not doing so (since the final user most likely won't be using them).

> Can you also squash together all the commits.

I'll be more careful with commits in the future... I never used Travis CI, so I was figuring out how it worked (same with tox). I also don't have python2.7 since it's EOF, so testing with it was a little bit tough.

Squashing commits can be done before merging the pull request: [About github's merge options](https://rietta.com/blog/github-merge-types/)

Thankss

**Comment by nap** [comment]
@GustavoPeredo I haven't forgotten. I'll get on it ASAP. Sorry about that.

**Pull request closed without merging**